### PR TITLE
feat: add teal-language-server support

### PR DESCRIPTION
**Pull request by svermeulen**

### CODE REVIEW

File: lua/lspconfig/server_configurations/teal_language_server.lua [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        "teal-language-server",
+        -- use this to enable logging in /tmp/teal-language-server.log
+        -- "logging=on", 
+     },
+     filetypes = {
+        "teal",
+        -- "lua", -- Also works for lua, but you may get type errors that cannot be resolved within lua itself
+     },
+     root_dir = util.root_pattern("tlconfig.lua", ".git"),
+     settings = {},
```
Comment (mjlbach) [inline lua/lspconfig/server_configurations/teal_language_server.lua:15]: Does settings need to be an empty table?

**Comment by mjlbach** [comment]
We typically shorten teal_language_server.lua -> teal_ls.lua

**Comment by mjlbach** [comment]
@svermeulen Do you have time to address the review? If so I will merge.

**Comment by svermeulen** [comment]
> @svermeulen Do you have time to address the review? If so I will merge.

Ok I've made the recommended changes

**Comment by mjlbach** [comment]
linter error (trailing blankspace)

**Comment by svermeulen** [comment]
Ok I think it may be fixed now

**Pull request merged**